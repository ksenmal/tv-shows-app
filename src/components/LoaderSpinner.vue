<template>
  <div class="loader-wrapper">
    <div class="loader"></div>
  </div>
</template>

<style scoped>
.loader-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 80vh;
}
.loader {
  border: 12px solid #f3f3f3; /* Light grey */
  border-top: 12px solid #3498db; /* Blue */
  border-radius: 50%;
  width: 80px;
  height: 80px;
  animation: spin 2s linear infinite, fadein 1s linear;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes fadein {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
</style>
