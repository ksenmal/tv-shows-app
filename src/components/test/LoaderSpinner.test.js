import { mount } from "@vue/test-utils";
import LoaderSpinner from "../LoaderSpinner.vue";
import { describe, expect, test } from "vitest";

describe("LoaderSpinner suite", () => {
  test("Snapshot", () => {
    const wrapper = mount(LoaderSpinner);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
