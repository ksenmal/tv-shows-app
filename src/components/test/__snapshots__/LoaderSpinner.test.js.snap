// Vitest Snapshot v1

exports[`LoaderSpinner suite > Snapshot 1`] = `
"<div class=\\"loader-wrapper\\" data-v-7afe7fb8=\\"\\">
  <div class=\\"loader\\" data-v-7afe7fb8=\\"\\"></div>
</div>"
`;
